store/postgres/src/deployment.rs

```diff
@@ -688,6 +688,24 @@ pub(crate) fn has_non_fatal_errors(
     .map_err(|e| e.into())
 }
 
+/// Looks for deterministic errors if the subgraph has the non-fatal errors
+/// feature enabled.
+pub(crate) fn has_non_fatal_errors(
```

Comment:
Introducing this function is no longer just a refactor but new behaviour. The only call site I found was here

https://github.com/graphprotocol/graph-node/blob/950993cf7c56f72800ccc65c10a5441904f2743a/graphql/src/store/resolver.rs#L82

Where the intended semantics are `has_deterministic_errors`, because we do want to return `"indexing_error"` on subgraphs without the non-fatal errors feature which hit a determinstic error. It seems we don't have a test for this so I added one here #3788, you can test that it breaks with this PR.

Reply:
It does indeed fail 😞 

<img width="936" alt="スクリーンショット 2022-07-29 12 33 33" src="https://user-images.githubusercontent.com/15306309/181793981-5b3d6c19-2bdf-412b-9233-c9dac3d122ac.png">